PWGDQ/Core/VarManager.h

```diff
@@ -1593,6 +1595,34 @@ void VarManager::FillTrackCollision(T const& track, C const& collision, float* v
   }
 }
 
+template <uint32_t fillMap, typename T, typename C, typename M, typename P>
+void VarManager::FillTrackCollisionMatCorr(T const& track, C const& collision, M const& materialCorr, P const& propagator, float* values)
+{
+  if (!values) {
+    values = fgValues;
+  }
+  if constexpr ((fillMap & ReducedTrackBarrel) > 0 || (fillMap & TrackDCA) > 0) {
+    auto trackPar = getTrackPar(track);
+    std::array<float, 2> dca{1e10f, 1e10f};
+    std::array<float, 3> pVec = {track.px(), track.py(), track.pz()};
+    // trackPar.propagateParamToDCA({collision.posX(), collision.posY(), collision.posZ()}, fgMagField, &dca);
+    o2::base::Propagator::Instance()->propagateToDCABxByBz({collision.posX(), collision.posY(), collision.posZ()}, trackPar, 2.f, materialCorr, &dca);
```

Comment:
you should not use "o2::base::Propagator::Instance()" here, as you have an argument "propagator". please call "propagator->PropagateToDCABxByBz". otherwise, I am afraid that you propagate tracks without magnetic field.

Reply:
For the first comment on px,py,pz, the recalculated values are not used indeed. The invariant mass and ptee are also not recalculated (we would need a similar function as VarManager::FillPairPropagateMuon and use it in the pairing). Maybe I would check before your other points.

Reply:
Maybe a comment/question from my side. Do we expect a significant change in the momentum? This should have very little sensitivity to the most inner point and should be mostly determined by the TPC and ITS. Especially with the requirement on the 1st spd hit the lever arm doesn't really change much.
However, we should of course implement this right.

Reply:
Yes, this was my point....I am not sure for the barrel that we expect anyway a big difference. Therefore I would first check how big is the difference and then decide maybe.

Reply:
With a small test using 
`o2::base::Propagator::Instance()->propagateToDCABxByBz({collision.posX(), collision.posY(), collision.posZ()}, trackPar, 2.f, materialCorr, &dca);`
and
`propagator->propagateToDCABxByBz({collision.posX(), collision.posY(), collision.posZ()}, trackPar, 2.f, materialCorr, &dca);`
 assuming the magnetic field is propagated like this correctly, I don't see a difference.